src/main/java/com/google/api/codegen/config/GapicProductConfig.java

```diff
@@ -224,16 +228,23 @@ public static GapicProductConfig create(
       clientPackageName = settings.getPackageName();
     }
 
+    // Collect the interfaces (clients) and methods that we will generate on the surface.
+    // Not all methods defined in the protofiles will be generated on the surface.
+    ImmutableMap<Interface, InterfaceConfigProto> interfacesMap =
+        createInterfacesMap(
```

Comment:
Here we determine the ordered set of clients and methods to generate. These sets are then passed down further into the pipeline, so we don't have to pass around a isGapicConfigPresent flag any more.

The invoked methods `createInterfacesMap` and `createMethodsMap` should only ever be called here.